docs/assemblies/assembly-overview.adoc

```diff
@@ -15,6 +15,21 @@ It takes on the responsibility of receiving, filtering, and forwarding messages.
 
 An API provides a convenient means for implementing custom logic within the proxy.
 
+[id='assembly-overview-compatability{context}']
+== Compatibility
+
+[id='assembly-overview-compatability-api{context}']
+=== APIs
+Kroxylicious follows https://semver.org/#semantic-versioning-200[Semantic Versioning] rules. While we are still in the `0.y.z` phase we still take API compatability very seriously, we aim to provide at least two minor releases between deprecation and the removal of that deprecated item.
+
+We also consider our configuration model to be a "public API", and thus it will follow the same Semantic Versioning and deprecation rules.
+
+Kubernetes Custom Resources are obviously considered to be a public API, and we will make every effort to evolve them in line with Kubernetes best practice.
```

Comment:
We should link to that those Kube best practices are. We should also be explicit that 

* The CR APIs are currently Alpha and so come with out any compatibility guarantees between releases. 
* the CR APIs are versioned separately (e.g. we might get to 1.0 proxy while still having alpha or beta CR APIs)

Reply:
> The CR APIs are currently Alpha and so come with out any compatibility guarantees between releases.

We could make the same argument for our API as we are still 0.X but we have chosen not to do so, why would we take a different path with the k8s CR APIs? I was expecting that while Alpha API's allow us to make changes at will we'd follow the same practices we do for the other APIs.